File: Server Integration/Python/rtchooks.py
import logging

# #########################################################################
# RTC HOOKS

def receiveNode(token, room, collectionName, nodeName, configuration):
  logging.info("receiveNode %s %s %s %s %s" % (token, room, collectionName, nodeName, configuration))

def receiveNodeConfiguration(token, room, collectionName, nodeName, configuration):
  logging.info("receiveNodeConfiguration %s %s %s %s %s" % (token, room, collectionName, nodeName, configuration))

def receiveNodeDeletion(token, room, collectionName, nodeName ):
  logging.info("receiveNodeDeletion %s %s %s %s" % (token, room, collectionName, nodeName))

def receiveItem(token, room, collectionName, item):
  logging.info("receiveItem %s %s %s %s" % (token, room, collectionName, item))

def receiveItemRetraction(token, room, collectionName, nodeName, item):
  logging.info("receiveItemRetraction %s %s %s %s %s" % (token, room, collectionName, nodeName, item))

def receiveUserRole(token, room, collectionName, nodeName, userID, role):
  logging.info("receiveUserRole %s %s %s %s %s %s" % (token, room, collectionName, nodeName, userID, role))

services = {
  'RTCHOOKS.receiveNode'              : receiveNode,
  'RTCHOOKS.receiveNodeConfiguration' : receiveNodeConfiguration,
  'RTCHOOKS.receiveNodeDeletion'      : receiveNodeDeletion,
  'RTCHOOKS.receiveItem'              : receiveItem,
  'RTCHOOKS.receiveItemRetraction'    : receiveItemRetraction,
  'RTCHOOKS.receiveUserRole'          : receiveUserRole,
}

# #########################################################################
# MAIN APPLICATION

if __name__ == '__main__':
  from pyamf.remoting.gateway.wsgi import WSGIGateway
  from wsgiref import simple_server

  logfile = "rtc.log"

  # uncommment to log to console
  #logfile = None

  logging.basicConfig(filename=logfile, level=logging.INFO, format="%(asctime)s - %(message)s")

  gw = WSGIGateway(services)

  httpd = simple_server.WSGIServer(
    ('0.0.0.0', 8000), 
    simple_server.WSGIRequestHandler,
  )
  httpd.set_app(gw)

  try:
    # open for business
    sa = httpd.socket.getsockname()
    print "Running RTCHOOKS gateway on", sa[0], "port", sa[1], "..."
    httpd.serve_forever()
  except KeyboardInterrupt:
    pass

File: Sample Applications/Server Integration/Java/Gateway/tomcat/webapps/samples/WEB-INF/src/flex/samples/LCCS/RTCHOOKS.java
package flex.samples.LCCS;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class RTCHOOKS {

	public void receiveNode(String token, String roomName, String collectionName, String nodeName, Map config) {
		
		System.out.println("ReceiveNode: " + token + " " + roomName + " " + collectionName + " " + nodeName);
		
		

	}

	public void receiveNodeDeletion(String token, String roomName, String collectionName, String nodeName) {
		
		System.out.println("receiveNodeDeletion: " + token + " " + roomName + " " + collectionName + " " + nodeName);
		
	}
	
	public void receiveItem(String token, String roomName, String collectionName, Map itemObj) {

		System.out.println("receiveItem: " + token + " " + roomName + " " + collectionName);
	}

	public void receiveItemRetraction(String token, String roomName, String collectionName, String nodeName, Map itemObj){
		
		System.out.println("receiveItemRetraction: " + token + " " + roomName + " " + collectionName + " " + nodeName);
		
	
	}

	public void receiveNodeConfiguration(String token, String roomName, String collectionName, String nodeName, Map config) {

		System.out.println("receiveNodeConfiguration: " + token + " " + roomName + " " + collectionName + " " + nodeName);
		
	}
	
	public void receiveUserRole(String token, String roomName, String collectionName, String nodeName, String userID, int role) {
		
		System.out.println("receiveUserRole: " + token + " " + roomName + " " + collectionName + " " + nodeName + " " + userID + " " + role);
	}

}
